mod_form.php

```diff
@@ -56,9 +56,38 @@ public function definition() {
             $nexturl = $PAGE->url;
             zoom_fatal_error($errstring, 'mod_zoom', $nexturl, $config->zoomurl);
         }
+        $isnew = empty($this->_cm);
+
+        /**
+         * @var $scheduleusers array Array of emails and proper names of Moodle users in this course that can add Zoom meetings, and the user can schedule.
+        */
+        $scheduleusers = [];
+        $scheduleusers[$USER->email] = get_string('scheduleforself', 'zoom');
+
+        // This will either be false (they can't) or the list of users they can schedule.
+        $canschedule = $service->get_schedule_for_users($USER->email);
+        if (!empty($canschedule)) {
+            // Get list of schedule for users if supported.
+            // List of users who can use Zoom mod in this class.
+            // We can use $this->context as this is set either to the constructor or the activity's context
+            // if it is an existing activity. This is good as the cap could be overridden in the activity permissions.
+            $moodleusers = get_enrolled_users($this->context, 'mod/zoom:addinstance', 0, 'u.*', 'lastname');
+            foreach ($canschedule as $zoomuser) {
+                $zoomid = $zoomuser->id;
+                $zoomemail = $zoomuser->email;
+                foreach ($moodleusers as $muser) {
+                    if ($muser->email === $USER->email) {
+                        // They're already in the $scheduleusers array in 1st position with "Schedule for self"
+                        continue;
+                    }
+                    if ($muser->email === $zoomemail) {
+                        $scheduleusers[$muser->email] = fullname($muser);
+                    }
```

Comment:
We don't need to loop through the rest of the users if we found the one that we want.
```suggestion
                        break;
                    }
```